### PR TITLE
Ensure read access to the run image selected by extensions

### PR DESCRIPTION
**Pull request by pbusko**
### Summary

When an extension switches the run image to one listed in the `run.toml`, the read access check is not triggered, as it is done in the analyze phase. This change will iterate over the requested run image and it's mirrors, to select the first accessible one.

#### Release notes

---

### Related

Resolves #___

---

### CODE REVIEW

**Review by jabrown85 — APPROVED** [review]

**Comment by loewenstein-sap** [comment]
@jabrown85 With your approval, what's missing to get it merged and released? Do we need @natalieparellano's approval in addition?

**Comment by jabrown85** [comment]
@loewenstein-sap yeah - I think I'd like a second set of eyes on this. What comes to mind now is that this phase may need registry auth, which isn't typically a thing in the untrusted phase executions. `CNB_REGISTRY_AUTH` is set for exporter/analyzer/restorer but is not mounted for builder/detector. I'm not sure that value should be set but that also doesn't mean generator phase shouldn't check access for what should be assumed to be unauthenticated image access.

There are other registry related settings like `CNB_INSECURE_REGISTRIES`, `CNB_USE_LAYOUT`, `CNB_LAYOUT_DIR` to think about as well.

**Comment by natalieparellano** [comment]
> What comes to mind now is that this phase may need registry auth, which isn't typically a thing in the untrusted phase executions.

This is true. Can we move this check to the restore phase?

**Comment by loewenstein-sap** [comment]
@natalieparellano ^^

**Comment by natalieparellano** [comment]
https://github.com/buildpacks/lifecycle/pull/1338 would have made this PR much easier to test :) unfortunately, that PR is blocked on me having some bandwidth to add units.

Apologies for being slow here, I've just returned from vacation. I'll take a look today.

File: cmd/lifecycle/restorer.go
Comment (natalieparellano) [inline cmd/lifecycle/restorer.go:53]: I am sorry to raise the specter of another Platform API version bump, but I think we need to gate this on Platform 0.14 also (an raise another spec PR).

**Review by natalieparellano — APPROVED** [review]

**Comment by natalieparellano** [comment]
I may have been too hasty in merging this one and https://github.com/buildpacks/lifecycle/pull/1346, as we're due for a go version / dependencies patch and this should be in the next minor...

**Pull request merged**
